crates/exex/exex/src/context.rs

```diff
@@ -92,4 +93,29 @@ impl<Node: FullNodeComponents> ExExContext<Node> {
     pub fn task_executor(&self) -> &TaskExecutor {
         self.components.task_executor()
     }
+
+    /// Replaces notifications stream with a stream of notifications without a head.
+    ///
+    /// Consumes the type and returns a new instance of [`ExExContext`].
+    pub fn with_notifications_without_head(self) -> Self {
+        Self { notifications: self.notifications.without_head(), ..self }
+    }
+
+    /// Replaces notifications stream with a stream of notifications with a head using the
+    /// provided head.
+    ///
+    /// Consumes the type and returns a new instance of [`ExExContext`].
+    pub fn with_notifications_with_head(self, head: ExExHead) -> Self {
```

Comment:
this is a bit verbose, with_with_head

Reply:
https://github.com/paradigmxyz/reth/pull/11500#discussion_r1790105958